graph-api.yaml

```diff
@@ -0,0 +1,459 @@
+#
+# Copyright (C) 2024 Telicent Limited
+#
+openapi: 3.0.3
+info:
+  title: Graph Smart-Cache API
+  description: |
+    OpenAPI Specification for interacting with the Telicent Smart-Cache Graph Server.
+    
+    **Note:** This description is obviously configuration dependent with the endpoints below being enabled/disabled as the relevant deployment requires.
+  version: 0.82.8
+servers:
+  - url: http://localhost:3030
+    description: Local Fuseki server instance
+paths:
+  /{dataset}/query:
+    get:
+      summary: Execute a SPARQL Query
+      description: |
+        Submit a SPARQL query against a dataset. 
+        
+        **Note:** Depending on configuration there can be equivalent /get /read or /sparql endpoints with identical functionality.
+      parameters:
+        - $ref: '#/components/parameters/dataset'
+        - $ref: '#/components/parameters/sparqlQuery'
+      responses:
+        200:
+          description: Query executed successfully.
+        400:
+          description: Invalid query or bad request.
+        401:
+          description: Unauthorised query.
+        404:
+          description: Endpoint not found i.e. incorrect dataset.
+        500:
+          description: Internal Server Error i.e. the server failed to process the request.
+        503:
+          description: Service Unavailable, i.e. the server is still in the process of starting.
+  /{dataset}/update:
+    post:
+      summary: Execute a SPARQL Update
+      description: | 
+        Submit a SPARQL update against a dataset and, by means of the RDF Patches generated, also update the other Smart Caches.
+      parameters:
+        - $ref: '#/components/parameters/dataset'
+      requestBody:
+        required: true
+        content:
+          application/sparql-update:
+            schema:
+              type: string
+              description: SPARQL Update query string.
+      responses:
+        200:
+          description: Update executed successfully.
+        400:
+          description: Invalid query or bad request.
+        401:
+          description: Unauthorised query.
+        404:
+          description: Endpoint not found i.e. incorrect dataset.
+        500:
+          description: Internal Server Error i.e. the server failed to process the request.
+        503:
+          description: Service Unavailable, i.e. the server is still in the process of starting.
+  /{dataset}/shacl:
+    get:
+      summary: Retrieve SHACL shapes for a dataset
+      description: |
+        Retrieve the SHACL validation shapes associated with a specific dataset.
+        The shapes are returned in RDF format and can be used to validate data against the dataset's SHACL rules.
+      parameters:
+        - $ref: '#/components/parameters/dataset'
```

Comment:
We are missing a couple of parameters here. First there is a query parameter required called `graph` which tells the validation process which graph to look in by URI. If you are looking at the default graph the value is `default`.

There is also an optional parameter called `target` which can be used to target one specific resource in the graph by providing the URI (or prefixed resource name).

Example: `http://localhost:3031/knowledge/shacl?graph=default&target=iow:_T036_TS072_provider `

Reply:
Will add, I appreciate the example